pkg/lib/cockpit.d.ts

```diff
@@ -93,6 +93,49 @@ declare module 'cockpit' {
     function channel(options: ChannelOptions & { binary?: false; }): Channel<string>;
     function channel(options: ChannelOptions & { binary: true; }): Channel<Uint8Array>;
 
+    /* === cockpit.spawn ============================= */
+
+    interface DeferredPromise<T> extends Promise<T> {
```

Comment:
I guess this `DeferredPromise` stuff belongs in its own section, assuming we're going to have it exposed by other things that are not spawn...

But my question: do we have to expose it at all?  Can we encourage new code to avoid this stuff?

Reply:
It does belong in it's own section, just here for convenience

We absolutely do not want to promote this, but our public API's return it. We would need to convert our API's to real promises that would be great but `progress` / `notify` we can't support.